src/network/networkd-link.c

```diff
@@ -1183,8 +1183,11 @@ void link_check_ready(Link *link) {
                         return;
                 }
 
-                if ((link_dhcp4_enabled(link) || link_dhcp6_enabled(link)) && set_isempty(link->addresses)) {
-                        log_link_debug(link, "%s(): DHCP4 or DHCP6 is enabled but no address is assigned yet.", __func__);
+                if ((link_dhcp4_enabled(link) || link_dhcp6_enabled(link) || link_dhcp6_pd_is_enabled(link)) &&
+                    !link->dhcp_address && !set_isempty(link->dhcp6_addresses) &&
```

Comment:
Isn't the condition inverted here? I'd expect `!link->dhcp_address && set_isempty(link->dhcp6_addresses) && ...`

Reply:
Ouch! Fixed.